File: 14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ''
        index = 0
        while True:
            for s in strs:
                if len(s) <= index or s[index] != strs[0][index]:
                    return strs[0][:index]
            index += 1


# s = Solution()
# print s.longestCommonPrefix(['abc', 'abccd'])

File: go/14.go

package main

import "fmt"

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {return ""}
	index := 0
	for {
		for _, str := range strs {
			if index >= len(str) {
				return str[:index]
			} else if str[index] == strs[0][index]{
				continue

			} else {
				return str[:index]
			}
		}
		index += 1
	}

}


func main() {
	strs := []string {"flower","flow","flight"}
	rs := longestCommonPrefix(strs)
	fmt.Println(rs)
}